langchain/llms/base.py

```diff
@@ -1,14 +1,26 @@
 """Base interface for large language models to expose."""
 from abc import ABC, abstractmethod
-from typing import Any, List, Mapping, Optional
+from typing import Any, List, Mapping, Optional, NamedTuple
+
+
+class LLMResult(NamedTuple):
+    """Class that contains all relevant information for an LLM Result."""
+    result: str
+    log_probs: List[float]
+    llm_output: dict
+    """For arbitrary LLM provider specific output."""
+
 
 
 class LLM(ABC):
     """LLM wrapper should take in a prompt and return a string."""
 
-    @abstractmethod
+    def generate(self, prompt: str, stop: Optional[List[str]] = None) -> LLMResult:
```

Comment:
What do you think of generate supporting multiple completions (batching)? Could be a list of LLMResults

Reply:
yeah great call